make install should create intermediate directories

When running `make install`, before copying files to the destination directory the whole path should be created:

```
$ make PREFIX=$HOME/opam install 
make opam-install opam-mk-repo-install opam-repo-convert-0.3-install
make[1]: Entering directory `/home/chambart/OcamlPro/git/opam'
cp _obuild/opam/opam.asm /home/chambart/opam/bin/opam
cp: cannot create regular file `/home/chambart/opam/bin/opam': No such file or directory
make[1]: *** [opam-install] Error 1
make[1]: Leaving directory `/home/chambart/OcamlPro/git/opam'
make: *** [install] Error 2
```

--- a/src/core/opamSystem.ml
+++ b/src/core/opamSystem.ml
@@ -591,7 +591,7 @@ let download ~overwrite ~filename:src ~dst:dst =
     really_download ~overwrite ~src ~dst
 
 let patch p =
-  let max_trying = 4 in
+  let max_trying = 5 in
   if not (Sys.file_exists p) then
     internal_error "Patch file %S not found." p;
   let patch ~dryrun n =

